trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -611,10 +613,15 @@ private Map<String, IFunctionModelHandler> getFunctionModels() {
 				(main, node, loc, name) -> handleVerifierNonDet(main, loc, new CPrimitive(CPrimitives.USHORT)));
 
 		/** from assert.h */
+		/** C standard library functions (from assert.h) to define the 'assert' macro */
 		fill(map, "__assert_fail", this::handleAssertFail);
 		fill(map, "__assert_func", this::handleAssertFail);
 		// TODO: This should not occur in the preprocessed file, but we handle it for now
 		fill(map, "assert", this::handleAssert);
+		/** C11 static assertion (C language keyword, deprecated in C23) */
+		fill(map, "_Static_assert", this::handleStaticAssert);

```

Comment:
Just to make sure: Are these the functions that occur in the preprocessed files?

Reply:
No, these functions are C language keywords defined by the C standard. Currently, we cannot handle C standard-specific keywords in the CDT parser, so we treat them somewhat hacky as functions.

Reply:
Okay, but i mean, if I preprocess a file, it still contains `_Static_assert` and `static_assert` afterwards, right? So these are not macros that are replaced (like `assert(...)`). They are of course specific keywords, but the "look like" function, so our handling should be fine.

Reply:
A C file still contains both static assertion keywords after preprocessing. As you mentioned, our handling should be fine here.